editor/animation_track_editor_plugins.cpp

```diff
@@ -1106,15 +1128,30 @@ void AnimationTrackEditTypeAudio::gui_input(const Ref<InputEvent> &p_event) {
 
 	Ref<EditorUndoRedoManager> &undo_redo = EditorNode::get_undo_redo();
 	if (len_resizing && mb.is_valid() && !mb->is_pressed() && mb->get_button_index() == MouseButton::LEFT) {
-		float ofs_local = -len_resizing_rel / get_timeline()->get_zoom_scale();
+		if (len_resizing_rel == 0 || len_resizing_index < 0) {
+			len_resizing = false;
+			return;
+		}
+
 		if (len_resizing_start) {
```

Comment:
```suggestion
		if (len_resizing_start) {
			float ofs_local = len_resizing_rel / get_timeline()->get_zoom_scale();
			float prev_ofs = get_animation()->audio_track_get_key_start_offset(get_track(), len_resizing_index);
			float prev_time = get_animation()->track_get_key_time(get_track(), len_resizing_index);
			float new_ofs = prev_ofs + ofs_local;
			float new_time = prev_time + ofs_local;
			if (prev_time != new_time) {
				undo_redo->create_action(TTR("Change Audio Track Clip Start Offset"));

				undo_redo->add_do_method(get_animation().ptr(), "track_set_key_time", get_track(), len_resizing_index, new_time);
				undo_redo->add_undo_method(get_animation().ptr(), "track_set_key_time", get_track(), len_resizing_index, prev_time);

				undo_redo->add_do_method(get_animation().ptr(), "audio_track_set_key_start_offset", get_track(), len_resizing_index, new_ofs);
				undo_redo->add_undo_method(get_animation().ptr(), "audio_track_set_key_start_offset", get_track(), len_resizing_index, prev_ofs);

				undo_redo->commit_action();
			}
		} else {
			float ofs_local = -len_resizing_rel / get_timeline()->get_zoom_scale();
			float prev_ofs = get_animation()->audio_track_get_key_end_offset(get_track(), len_resizing_index);
			float new_ofs = prev_ofs + ofs_local;
			if (prev_ofs != new_ofs) {
				undo_redo->create_action(TTR("Change Audio Track Clip End Offset"));
				undo_redo->add_do_method(get_animation().ptr(), "audio_track_set_key_end_offset", get_track(), len_resizing_index, new_ofs);
				undo_redo->add_undo_method(get_animation().ptr(), "audio_track_set_key_end_offset", get_track(), len_resizing_index, prev_ofs);
				undo_redo->commit_action();
			}
		}
```
The operation should be canceled.

Reply:
Didn't really understand what are you suggesting here

Reply:
Due to github's suggestion formatting, this suggestion has been wrote here, but the old unnecessary lines below from that need to be deleted at your discretion.